RHIDP-1454: Document TechDocs plugin (#294)
redhat-developer/red-hat-developers-documentation-rhdh: modules/getting-started/con-techdocs-config-cicd.adoc

> [id="con-techdocs-config-cicd_{context}"]
> 
> = Configuring CI/CD to generate and publish TecDocs sites

Reviewer: ```suggestion
// Module included in the following assemblies:
//
// * assemblies/assembly-techdocs-plugin.adoc

:_mod-docs-content-type: CONCEPT
[id="con-techdocs-config-cicd_{context}"]
= Configuring CI/CD to generate and publish TecDocs sites
```

> TechDocs reads the static generated documentation files from a cloud storage bucket, such as OpenShift Data Foundation. The documentation site is generated on the CI/CD workflow associated with the repository containing the documentation files. You can generate docs on CI and publish to a cloud storage using the `techdocs-cli` tool.

Reviewer: ```suggestion
TechDocs reads the static generated documentation files from a cloud storage bucket, such as {odf-name}. The documentation site is generated on the CI/CD workflow associated with the repository containing the documentation files. You can generate docs on CI and publish to a cloud storage using the `techdocs-cli` CLI tool.
```

You can use the following example to create a script for TechDocs publication:

> [source]

Reviewer: ```suggestion
[source,shell]
```

> Supported CI providers include GitHub Actions, CircleCI, or Jenkins. You can use the following example script to create a script for TechDocs publication:

Reviewer 2: Where is this coming from? We don't provide support for any of the CI providers.
Almost any  CI system can be used to generate techdocs.

Author: This content came from [this link](documented in https://backstage.io/docs/features/techdocs/configuring-ci-cd) that you shared. If it is not accurate for RHDH. Do we need to remove or change anything beyond _Supported CI providers include GitHub Actions, CircleCI, or Jenkins_?

Reviewer 2: the origin document makes it clear that the listed CI is not the only CI that you can use. Maybe I'm translating it wrong, but to me, it looks like the 3 listed CI providers are the only one that you can use. 

We should also be careful about using term "supported" as it might look like this is something that Red Hat provides support for.

Author: I now understand that it was probably also the word "including" that was causing the issue. I removed it. Rather than supported, we can say "available" or "compatible". The other option is to remove them altogether.

----
# Prepare
REPOSITORY_URL='https://github.com/org/repo'
git clone $REPOSITORY_URL
cd repo

# Install @techdocs/cli, mkdocs and mkdocs plugins
npm install -g @techdocs/cli
pip install "mkdocs-techdocs-core==1.*"

# Generate
techdocs-cli generate --no-docker

# Publish
techdocs-cli publish --publisher-type awsS3 --storage-name <bucket/container> --entity <Namespace/Kind/Name>
----

> The TechDocs workflow starts the CI when a user makes changes in the repository containing the documentation files. You can configure the workflow to start only when files inside the `docs/` directory or `mkdocs.yml` are changed.

Reviewer: Do we show users how to configure the workflow this way? It would be good to include that here.
